data-products/src/main/scala/org/sunbird/analytics/audit/ScoreMetricMigrationJob.scala

```diff
@@ -97,10 +127,51 @@ object ScoreMetricMigrationJob extends optional.Application with IJob with BaseR
     aggregation.toList.flatten.toMap
   }
 
-  def getBestScoreRecordsDF(assessmentDF: DataFrame): DataFrame = {
-    val df = Window.partitionBy("user_id", "batch_id", "course_id", "content_id").orderBy(desc("total_score"))
-    assessmentDF.withColumn("rownum", row_number.over(df)).where(col("rownum") === 1).drop("rownum")
-      .withColumn("batchid", concat(lit("cb:"), col("batch_id"))).drop("batch_id")
+  def mergeAggDetailListValues(): Seq[Seq[String]] => List[String] = (aggregation: Seq[Seq[String]]) => {
+    aggregation.toList.flatten
   }
 
+  def mergeAggDetailsCol(): (Seq[String], List[String]) => List[String] = (aggDetails: Seq[String], newAggDetails: Seq[String]) => {
+    val filteredAggDetails = newAggDetails.foldLeft(List[String]())((finalResult, aggDetail: String) => {
+      val aggDetailMap = JSONUtils.deserialize[Map[String, AnyRef]](aggDetail)
+      val existingAgg = aggDetails.filter(row => {
+        aggDetailMap.get("attempt_id").get == JSONUtils.deserialize[Map[String, AnyRef]](row).get("attempt_id").get
+      })
+      if (existingAgg.isEmpty) {
+        finalResult :+ aggDetail
+      } else finalResult
+    })
+
+    aggDetails.toList ++ filteredAggDetails
+  }
+
+
+
+  /**
+   * Get the best score out of the attempts and accumulates all atttempts detail in one list
+   * @param assessmentDF AssessmentAggregator records DF
+   * @param metrics_type
+   *
+   * @return DataFrame
+   * Example Output:
+   * +--------+-------------------------+-----------------------+-----------+---------------+-----------------------------------------------------------------------------------------------------------------------------+------------+
+   * |user_id |course_id                |content_id             |total_score|total_max_score|migrating_agg_details                                                                                                        |batchid     |
+   * +--------+-------------------------+-----------------------+-----------+---------------+-----------------------------------------------------------------------------------------------------------------------------+------------+
+   * |user-012|do_1130293726460805121168|do_11307593493010022419|10.0       |15.0           |[{"max_score":15.0,"score":10.0,"type":"attempt_metrics","attempt_id":"attempat-001","content_id":"do_11307593493010022419"}]|cb:batch-002|
+   * +--------+-------------------------+-----------------------+-----------+---------------+-----------------------------------------------------------------------------------------------------------------------------+------------+
+   */
+  def getBestScoreAggDetailsDF(assessmentDF: DataFrame, metrics_type: String): DataFrame = {
+    assessmentDF.groupBy("user_id", "batch_id", "course_id", "content_id").agg(
+      max("total_score").as("total_score"),
```

Comment:
Please rename it to "best_score" - `max("total_score").as("best_score")` so it will easy to know later